server/data_common/matrix_loader.py

```diff
@@ -122,11 +126,12 @@ def data_adaptor(self, location, app_config):
                 loader = MatrixDataLoader(location, app_config=app_config)
                 cache_item = MatrixDataCacheItem(loader)
                 self.datasets[location] = (cache_item, last_accessed)
-        try:
-            data_adaptor = cache_item.acquire(app_config)
-            yield data_adaptor
-        finally:
-            cache_item.release()
+
+            try:
+                data_adaptor = cache_item.acquire(app_config)
+                yield data_adaptor
```

Comment:
yielding will leave the lock held, which single-threads any caller that uses this as a generator.

do we actually use it this way?  It looks like the two invocations in app.py use it differently (one within a `with`, one just as a function).  I believe it will single thread callers through the first due to the fact that yield does not release the lock.